File: frontend/src/types/building.ts
// 楼宇管理类型定义

export interface Building {
  id: number
  buildingNo: string
  buildingName: string
  buildingType: number
  totalFloors: number
  location?: string
  constructionYear?: number
  managerId?: number
  managerName?: string
  description?: string
  status: number
  roomCount?: number
  buildingTypeName?: string
  // 组织单元关联字段
  orgUnitIds?: number[]
  orgUnitNames?: string | string[]  // 后端返回字符串(GROUP_CONCAT),前端需要转换为数组
  createdAt?: string
  updatedAt?: string
  createdBy?: number
  updatedBy?: number
  deleted?: number
}

export interface BuildingQueryParams {
  pageNum: number
  pageSize: number
  buildingNo?: string
  buildingName?: string
  buildingType?: number
  status?: number
}

export interface BuildingFormData {
  buildingNo: string
  buildingName: string
  buildingType: number
  totalFloors: number
  location?: string
  constructionYear?: number
  managerId?: number
  description?: string
  status: number
  // 组织单元关联
  orgUnitIds?: number[]
}

// 楼宇类型选项
export const buildingTypeOptions = [
  { label: '教学楼', value: 1 },
  { label: '宿舍楼', value: 2 },
  { label: '办公楼', value: 3 }
]

// 状态选项
export const buildingStatusOptions = [
  { label: '启用', value: 1 },
  { label: '停用', value: 0 }
]

File: backend/src/main/java/com/school/management/entity/Building.java
package com.school.management.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 楼宇实体
 *
 * @author system
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("buildings")
public class Building extends BaseEntity {

    /**
     * 楼号(纯数字,如: 1, 2, 3)
     */
    private String buildingNo;

    /**
     * 楼宇名称
     */
    private String buildingName;

    /**
     * 楼宇类型: 1-教学楼, 2-宿舍楼, 3-办公楼
     */
    private Integer buildingType;

    /**
     * 总楼层数
     */
    private Integer totalFloors;

    /**
     * 地理位置
     */
    private String location;

    /**
     * 建造年份
     */
    private Integer constructionYear;

    /**
     * 楼宇描述
     */
    private String description;

    /**
     * 状态: 0-停用, 1-启用
     */
    private Integer status;

    /**
     * 房间数量(非数据库字段)
     */
    @TableField(exist = false)
    private Integer roomCount;

    /**
     * 楼宇类型名称(非数据库字段)
     */
    @TableField(exist = false)
    private String buildingTypeName;

    /**
     * 关联组织单元ID列表(非数据库字段)
     */
    @TableField(exist = false)
    private java.util.List<Long> orgUnitIds;

    /**
     * 关联组织单元名称列表(非数据库字段) - SQL返回的是逗号分隔的字符串
     */
    @TableField(exist = false)
    private String orgUnitNames;

    /**
     * 获取楼宇类型名称
     */
    public String getBuildingTypeName() {
        if (buildingTypeName != null) {
            return buildingTypeName;
        }
        if (buildingType == null) {
            return "";
        }
        switch (buildingType) {
            case 1:
                return "教学楼";
            case 2:
                return "宿舍楼";
            case 3:
                return "办公楼";
            default:
                return "未知";
        }
    }
}
